src/nvim/buffer_updates.c

```diff
@@ -366,10 +388,16 @@ void buf_updates_changedtick(buf_T *buf)
       // next argument is b:changedtick
       args.items[1] = INTEGER_OBJ(buf_get_changedtick(buf));
 
+      pos_T save_cursor;
+      if (curwin != NULL) {
+        save_cursor = curwin->w_cursor;
+      }
       textlock++;
-      Object res = nlua_call_ref(cb.on_changedtick, "changedtick",
-                                 args, true, NULL);
+      Object res = nlua_call_ref(cb.on_changedtick, "changedtick", args, true, NULL);
       textlock--;
+      if (curwin != NULL) {
+        curwin->w_cursor = save_cursor;
+      }
```

Comment:
lots of common code in all of these functions. do we need a `buf_updates_prepare`/`..._restore` util functions? Or should we add a `textlock` param to `nlua_call_ref()` ?